plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/SchemaInitializer.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.trino.testing.QueryRunner;
+import io.trino.tpch.TpchTable;
+
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import static io.trino.plugin.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static io.trino.testing.QueryAssertions.copyTpchTables;
+import static java.util.Objects.requireNonNull;
+
+public class SchemaInitializer
+        implements Consumer<QueryRunner>
+{
+    private final String schemaName;
+    private final Map<String, String> schemaProperties;
+    private final Iterable<TpchTable<?>> clonedTpchTables;
+
+    private SchemaInitializer(String schemaName, Map<String, String> schemaProperties, Iterable<TpchTable<?>> tpchTablesToClone)
+    {
+        this.schemaName = requireNonNull(schemaName, "schemaName is null");
+        this.schemaProperties = requireNonNull(schemaProperties, "schemaProperties is null");
+        this.clonedTpchTables = requireNonNull(tpchTablesToClone, "tpchTablesToClone is null");
+    }
+
+    public String getSchemaName()
+    {
+        return schemaName;
+    }
+
+    @Override
+    public void accept(QueryRunner queryRunner)
+    {
+        String schemaProperties = this.schemaProperties.entrySet().stream()
+                .map(entry -> entry.getKey() + " = " + entry.getValue())
+                .collect(Collectors.joining(", ", " WITH ( ", " )"));
+        queryRunner.execute("CREATE SCHEMA IF NOT EXISTS " + schemaName + (this.schemaProperties.size() > 0 ? schemaProperties : ""));
+        copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, queryRunner.getDefaultSession(), clonedTpchTables);
+    }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static class Builder
+    {
+        private String schemaName = "tpch";
+        private Map<String, String> schemaProperties = ImmutableMap.of();
+        private Iterable<TpchTable<?>> cloneTpchTables = ImmutableSet.of();
+
+        public Builder withSchemaName(String schemaName)
```

Comment:
This needs to be in line with what you pass to `Builder` constructor.
Unless you use helper `createIcebergQueryRunner` which takes `SchemaInitializer` the responsibility to set it up correctly is put on the user of IceberQueryRUnner.Builder.

Maybe you can drop this method at all and always take the schema name from session, which is created in Builder constructor.

Then you have single source of truth and user does not have to be smart about usage.

Reply:
cc @alexjo2144 for same topic in https://github.com/trinodb/trino/pull/10845#discussion_r816661410

Reply:
Fair point, i overseen that. Will remove schemaName from builder method and modify default session in setSchemaInitializer method.

Reply:
You cannot modify default schema easily, since it's already baked in into the builder. 
see https://github.com/trinodb/trino/pull/10845#discussion_r816661410